sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/SymmetricHashJoinStateManager.scala

```diff
@@ -269,18 +269,14 @@ class SymmetricHashJoinStateManager(
         // The backing store is arraylike - we as the caller are responsible for filling back in
         // any hole. So we swap the last element into the hole and decrement numValues to shorten.
         // clean
-        if (numValues > 1) {
+        if (index != numValues - 1) {
           val valuePairAtMaxIndex = keyWithIndexToValue.get(currentKey, numValues - 1)
           if (valuePairAtMaxIndex != null) {
```

Comment:
Just wondering, would `valuePairAtMaxIndex == null` happen in normal case? I can only imagine the case the iterator is not fully consumed and somehow numValues is not updated.

Reply:
I guess no, for normal case. Seems better to keep it.

Reply:
Yeah my preference is to fail the query (like raising internal error) on the case of `valuePairAtMaxIndex == null`, so that we can indicate the bad case in prior instead of encountering bad case after state is corrupted.

If we would only have the case the value is null as error case, still makes sense to fail the query even for the state created before the fix, but for safety, we could log warning message here. I'm just afraid I'm missing some "normal" case where the null value is doable.

Reply:
ah, missing latest comment. Let me add a log warning message here in a follow-up.